### PR TITLE
Split Utils

### PR DESCRIPTION
**Pull request by zdrve**
Splitting up the grab-bag "Utils" class into three:

 - string stuff
 - console / UI stuff
 - the rest

which turns out to make about three equal-sized parts.

Potentially a **BREAKING CHANGE** for anything using `Utils`.

### CODE REVIEW

File: lib/kennel/string_utils.rb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+          .gsub(/^-|-$/, "") # remove leading/trailing
+      end
+
+      def truncate_lines(text, to:, warning:)
+        lines = text.split(/\n/, to + 1)
+        lines[-1] = warning if lines.size > to
+        lines.join("\n")
+      end
+
+      def natural_order(name)
```
Comment (grosser) [inline lib/kennel/string_utils.rb:34]: idk about this one ... feels like the wrong place

**Review by grosser — COMMENTED** [review]
the added complexity from now having to find where things are is kinda meh
... especially when it's not super-clear like `color` could be in console or string
and then over time things might move and more breaks :(

... I really liked the grouping of the _case/parameterize methods though
... maybe step 1 is split it here but extend them all into `Utils` so maintenance is simpler but the UI stays the same

File: lib/kennel/console_utils.rb [hunk 2]
```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+module Kennel
+  module ConsoleUtils
```
Comment (grosser) [inline lib/kennel/console_utils.rb:3]: maybe name this `Console` ?
`Kennel::Console.ask?` is kinda nice

File: lib/kennel/string_utils.rb (hunk 1 above)
Comment (zdrve) [reply lib/kennel/string_utils.rb:34]: Good point, the others are all string->string functions and this one is something very different.

**Comment by zdrve** [comment]
> ... especially when it's not super-clear like color could be in console or string

True insofar as it's both a string transform, and to do with console stuff... the distinction I see being that StringUtils is more generally-useful stuff (e.g. transforming strings to go into Datadog monitor titles and messages and so on), whereas you would never use `color` in such a context.

> the added complexity from now having to find where things are is kinda meh
> and then over time things might move and more breaks :(

I don't really buy these arguments, if that's what they are. One could also argue that having a file / module which is three times as large and contains unrelated things is "added complexity", i.e. this PR _reduces_ complexity. Ultimately, it's a trade-off; I haven't, for example, split it into 15 or so modules with one method each ;-)

**Pull request merged**